polynote-runtime/src/main/scala/polynote/runtime/DataType.scala

```diff
@@ -22,7 +25,17 @@ final case class StructField(
 
 final case class StructType(
   fields: List[StructField]
-) extends DataType
+) extends DataType {
+  override def size: Int = if (fields.isEmpty || fields.forall(_.dataType.size >= 0)) {
+    fields.map(_.dataType.size).sum
+  } else -1
```

Comment:
if this `-1` is supposed to match the default implementation, maybe `super.size` would be better?

Reply:
It's not because `-1` is the default... it's because size < 0 means "unknown". So if any of the fields are unknown size, then the entire thing is unknown size. It matches the default only because the default size is unknown, so I'm not sure it's clearer to say `super.size`.

Reply:
Ah I see. I guess maybe a `val UNKNOWN = -1` constant in `DataType` would make that super clear but is probably not necessary.